lucet-runtime/lucet-runtime-internals/src/region/mmap.rs

```diff
@@ -100,6 +101,13 @@ impl RegionInternal for MmapRegion {
         {
             // eprintln!("setting none {:p}[{:x}]", *ptr, len);
             unsafe {
+                // MADV_DONTNEED is not guaranteed to clear pages on non-Linux systems
+                #[cfg(not(target_os = "linux"))]
+                {
+                    mprotect(*ptr, *len, ProtFlags::PROT_READ | ProtFlags::PROT_WRITE)
+                        .expect("mprotect succeeds during drop");
+                    memset(*ptr, 0, *len);
```

Comment:
should we set `PROT_NONE` again after this `memset`?

Reply:
that's unconditionally done a few lines down from here :)

Reply:
so it is, lol :+1: